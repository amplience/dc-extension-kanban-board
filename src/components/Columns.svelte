<script>
  import Chip from './Chip.svelte';
  import Card from './Card.svelte';
  import Count from './Count.svelte';
  import { dndzone } from 'svelte-dnd-action';
  import type { StatusWithContentItemCollection } from '../services/data/content-items';
  export let handleConsider: any;
  export let handleFinalize: any;
  export let statuses: Array<StatusWithContentItemCollection> = [];
</script>

<style lang="scss">
  .grid {
    box-sizing: border-box;
    display: grid;
    grid-template-rows: 1fr;
    grid-gap: 8px;
    height: 100%;
    width: 100%;
    padding: 0.5em;
  }
  .col {
    box-sizing: border-box;
    background: #eee;
    overflow-y: hidden;

    .content-item-wrap {
      overflow-y: auto;
      height: calc(100% - 90px);
    }
  }
</style>

<div
  class="grid"
  style="grid-template-columns:{Array(statuses.length + 1)
    .join('1fr ')
    .trim()}">
  {#each statuses as status (status.id)}
    <div class="col">
      <Chip color={status.color} label={status.label} />
      <Count
        total={status.contentItems.page.totalElements}
        count={status.contentItems.items.length} />
      <div
        class="content-item-wrap"
        use:dndzone={{ items: status.contentItems.items, type: 'content-items' }}
        on:consider={(e) => handleConsider(status.id, e)}
        on:finalize={(e) => handleFinalize(status.id, e)}>
        {#each status.contentItems.items as contentItem (contentItem.id)}
          <Card
            title={contentItem.label}
            subtitle={contentItem.contentType}
            footer="Last changed {contentItem.modified}" />
        {/each}
      </div>
    </div>
  {/each}
</div>
