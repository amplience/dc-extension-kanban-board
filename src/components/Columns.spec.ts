import { render } from '@testing-library/svelte';
import Columns from './Columns.svelte';

describe('columns component', () => {
  it('should render a columns component', () => {
    const handleConsider = jest.fn();
    const handleFinalize = jest.fn();
    const client = {
      dashboardSdk: {
        applicationNavigator: {
          openContentItem: jest.fn(),
        },
      },
    };

    const { container } = render(Columns, {
      handleConsider,
      handleFinalize,
      statuses: [
        {
          id: 'ID',
          backgroundColor: 'RED',
          color: 'BLUE',
          label: 'STATUS_LABEL',
          hydrated: true,
          contentItems: {
            page: {},
            statusId: 'STATUS_ID',
            items: [
              {
                id: 'ID',
                label: 'LABEL',
                schema: 'schema',
                lastModifiedDate: '2020-01-01T00:00:59Z',
              },
            ],
            totalElements: 1,
          },
        },
      ],
      client: client,
      contentTypeLookup: {
        schema: {
          settings: {
            label: 'SCHEMA LABEL',
          },
        },
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
