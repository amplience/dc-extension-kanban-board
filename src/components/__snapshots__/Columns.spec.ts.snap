// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`columns component should render a columns component 1`] = `
<div>
  <div
    class="grid svelte-1m4vmnb"
    style="grid-template-columns: 1fr;"
  >
    <div
      class="col svelte-1m4vmnb"
    >
      <div
        class="chip BLUE svelte-1yjwfb7"
        style="background-color: RED;"
      >
        <div
          class="truncate svelte-atm87y"
          title="STATUS_LABEL"
        >
          STATUS_LABEL
        </div>
      </div>
       
      <div
        class="counts-and-actions svelte-1m4vmnb"
      >
        <div
          class="count svelte-1aviz9u"
        >
          <span
            class="bold svelte-1aviz9u"
          >
            0
          </span>
          
  content item
          s
           
           
          (from last 7 days)
        </div>
      </div>
       
      <div
        aria-describedby="dnd-zone-active"
        aria-disabled="false"
        class="content-item-wrap svelte-1m4vmnb"
        role="list"
        tabindex="0"
      />
       
    </div>
  </div>
</div>
`;
